pkgs/build-support/docker/default.nix

```diff
@@ -292,7 +296,7 @@ rec {
 
       postMount = ''
         echo "Packing raw image..."
-        tar -C mnt --hard-dereference --sort=name --mtime="@$SOURCE_DATE_EPOCH" -cf $out .
+        tar -C mnt --hard-dereference --sort=name --mtime="@$SOURCE_DATE_EPOCH" -cf $out/layer.tar .
```

Comment:
How about adding a `touch $out/layerTarAtRoot`, then using `if [ -e layerTarAtRoot ]` in the `runWithOverlay` to decide whether to move it to the root or not.

Reply:
That would work, I can make the change, just waiting to see if there's any resolution to the above conversation.

Reply:
Let's make sure we always represent layers in the same way, instead of adding more conditionals.

Reply:
I've removed the conditionals by making fixing the original problem of `createEmptyImage` clutter in a better way.

Representing layers consistently doesn't really apply here.